src/ServicePulse.Host/vue/src/components/failedmessages/MessageView.vue

```diff
@@ -244,45 +281,47 @@ function formatXml(xml) {
   return string.trim();
 }
 
-function formatJson(json) {
+function formatJson(json: string) {
   return JSON.stringify(json, null, 2);
 }
 
-function togglePanel(panelNum) {
-  if (!failedMessage.value.notFound && !failedMessage.value.error) {
+function togglePanel(panelNum: number) {
+  if (!failedMessage.value?.notFound && !failedMessage.value?.error) {
```

Comment:
could this cause incorrect behaviour is failedMessage is null? Perhaps if (failedMessage.value && !failedMessage.value.notFound && !failedMessage.value.error)